tools/release-ci/pkg/buildrunner/runbuild.go

```diff
@@ -0,0 +1,211 @@
+package buildrunner
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+)
+
+var (
+	WaitForRunningTimeout = 10 * time.Minute
+	MaximumBuildDuration  = 4 * time.Hour
+)
+
+func RunBuild(resourceFile string) error {
+
+	if len(resourceFile) == 0 {
+		return errors.New("no build file specified")
+	}
+	if resourceFile == "-" {
+		tmp, err := ioutil.TempFile("", "")
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(tmp, os.Stdin)
+		if err != nil {
+			return err
+		}
+		tmp.Close()
+		resourceFile = tmp.Name()
+	}
+	result, err := runCmd("oc", "create", "-f", resourceFile)
+	var buildName = ""
+	if err == nil {
+		// Parse the build name from the result
+		buildName, err = extractSuccessfulBuildName(result)
+		if err != nil {
+			return err
+		}
+	} else {
+		buildName, err = extractExistingBuildName(result)
+		if err != nil {
+			return err
+		}
+	}
+	return followBuild(buildName)
+}
+
+func extractSuccessfulBuildName(str string) (string, error) {
+	createdBuildRegex := regexp.MustCompile("build \"([^\"]*)\" created")
```

Comment:
can't we `-o name` to not need to parse potentially unstable human readable text?

Reply:
We can parse the Yaml/JSON and get the name from there. I guess I was being lazy? I don't want to make an extra call to the server if I don't need to.

Reply:
Will fix